Tasks/VSMobileCenterTest/vsmobilecentertest.ts

```diff
@@ -0,0 +1,213 @@
+        } 
+        else if (framework === 'uitest') {
+            prepareRunner.arg(['--app-path', app]);
+            addStringArg('--build-dir', 'uitestBuildDir', true, prepareRunner); 
+            addStringArg('--store-file', 'uitestStoreFile',false, prepareRunner);
+            addStringArg('--store-password', 'uitestStorePass',false, prepareRunner);
+            addStringArg('--key-alias', 'uitestKeyAlias',false, prepareRunner);
+            addStringArg('--key-password', 'uitestKeyPass',false, prepareRunner);
+            addStringArg('--uitest-tools-dir', 'uitestToolsDir',false, prepareRunner);
+            addStringArg('--sign-info', 'signInfo', false, prepareRunner); 
+        }
+
+        // append user defined inputs
+        let prepareOpts = tl.getInput('prepareOpts', false);
+        if (prepareOpts) {
+            prepareRunner.line(prepareOpts);
+        }
+
+        if (debug) {
+            prepareRunner.arg('--debug');
+        }
+        prepareRunner.arg('--quiet');
+
+        // Get build inputs
+        let testRunner = tl.tool(cliPath);
+        testRunner.arg(['test', 'run', 'manifest']);
+        testRunner.arg(['--manifest-path', `${path.join(artifactsDir, 'manifest.json')}`]);
+        testRunner.arg(['--app-path', app, '--app', appSlug]);
+
+        addStringArg('--devices', 'devices', true, testRunner); 
+        addStringArg('--test-series', 'series', false, testRunner); 
+        addStringArg('--dsym-dir', 'dsymDir', false, testRunner); 
+        addBooleanArg('--async', 'async', testRunner); 
+
+        let locale: string = tl.getInput('locale', true);
+        if (locale === 'user') {
+            tl.debug('Use user defined locale.'); 
+            locale = tl.getInput('userDefinedLocale', true);
+        } 
+        testRunner.arg(['--locale', locale]);
+
+        let options: string = tl.getInput('opts', false);
+        if (options) {
+            testRunner.line(options);
+        }
+        if (debug) {
+            testRunner.arg('--debug');
+        }
+        testRunner.arg('--quiet');
+        if (credsType === 'serviceEndpoint'){
+            // add api key
+            let apiToken = getEndpointAPIToken('serverEndpoint');
+            testRunner.arg(['--token', apiToken]);
+        }
+
+        // login if necessary
+        if (credsType === 'inputs') {
+            await loginRunner.exec();
+            loggedIn = true;
+        }
+
+        // test prepare
+        await prepareRunner.exec();
+
+        // test run
+        await testRunner.exec();
+
+        tl.setResult(tl.TaskResult.Succeeded, tl.loc("Succeeded"));
+    } catch (err) {
+        tl.setResult(tl.TaskResult.Failed, `${err}`);
+    } finally {
+        if (tl.exist(cliPath) && loggedIn) {
+            // logout
+            let logoutRunner = tl.tool(cliPath);
+            logoutRunner.arg(['logout', '--quiet']); 
+            logoutRunner.exec();
```

Comment:
sure, I will.  This is clean up code so I didn't think we need to wait.  However it's safer if we do.